File: ride-service/src/main/java/com/example/rideservice/service/impl/TemperatureServiceImpl.java
package com.example.rideservice.service.impl;

import com.example.rideservice.service.TemperatureService;
import com.github.prominence.openweathermap.api.OpenWeatherMapClient;
import com.github.prominence.openweathermap.api.enums.UnitSystem;
import com.github.prominence.openweathermap.api.model.Temperature;
import com.github.prominence.openweathermap.api.request.weather.CurrentWeatherRequester;
import com.github.prominence.openweathermap.api.request.weather.single.SingleResultCurrentWeatherRequestTerminator;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class TemperatureServiceImpl implements TemperatureService {

    private final OpenWeatherMapClient client;

    @Override
    public double calculatePrice(String city) {
        CurrentWeatherRequester requester = client.currentWeather();

        SingleResultCurrentWeatherRequestTerminator terminator = requester.single()
                .byCityName(city)
                .unitSystem(UnitSystem.METRIC)
                .retrieve();

        Temperature temperature = terminator.asJava().getTemperature();
        return temperature.getValue();
    }
}

File: price-calculator-service/src/main/kotlin/org/example/pricecalculatorservice/service/impl/TemperatureServiceImpl.kt
package org.example.pricecalculatorservice.service.impl

import com.github.prominence.openweathermap.api.OpenWeatherMapClient
import com.github.prominence.openweathermap.api.enums.UnitSystem
import com.github.prominence.openweathermap.api.request.weather.CurrentWeatherRequester
import org.example.pricecalculatorservice.exception.SomethingWrongTemperatureServiceException
import org.example.pricecalculatorservice.service.TemperatureService
import org.springframework.stereotype.Service

@Service
class TemperatureServiceImpl(private val openweathermapClient: OpenWeatherMapClient) : TemperatureService {
    override fun getTemperatureByWeather(city: String): Double {


        try {

            val requester: CurrentWeatherRequester = openweathermapClient.currentWeather()
            val terminator = requester.single()
                .byCityName(city)
                .unitSystem(UnitSystem.METRIC)
                .retrieve()

            val temperature = terminator.asJava().temperature
            return temperature.value
        } catch (ex: Exception) {
            throw SomethingWrongTemperatureServiceException(ex.message.toString());
        }
    }
}